docker-compose.yml

```diff
@@ -0,0 +1,64 @@
+version: "3.9"
+
+services:
+  app:
+    build:
+      args:
+        user: opendor-dev
+        uid: 1000
+      context: ./
+      dockerfile: .docker/php/Dockerfile
+    restart: unless-stopped
+    working_dir: /var/www/
+    volumes:
+      - ./:/var/www
+    depends_on:
+      - db
+      - redis
+    networks:
+      - opendor
+
+  db:
+    image: postgres:13
+    restart: unless-stopped
+    environment:
+      POSTGRES_DB: ${DB_DATABASE}
+      POSTGRES_USER: ${DB_USERNAME}
+      POSTGRES_PASSWORD: ${DB_PASSWORD}
+    volumes:
+      - psql:/var/lib/postgresql/data
+    ports:
+      - "${DB_PORT:-5432}:5432"
```

Comment:
Would echo @Gummibeer's suggestion to use nonstandard ports.
In my case this docker image failed since I already have a local postgres instance running.

Reply:
As long as I'm able to adjust it I'm fine with default port on host. But there should be a way and documentation where I have to change that. As docker defaults to `.env` as well I'm not sure how that works combined with the Laravel `.env` 🤔

Reply:
in `.env` file use some other port for `DB_PORT` and it will use that port.

Reply:
Actually that's a great point!

I literally just needed to edit my local `.env` to change the port to `54320`.
Now when I use compose to start the stack it's on the proper port.

![Screen Shot 2021-07-16 at 12 01 14 PM](https://user-images.githubusercontent.com/619938/125976090-0d611cf2-9965-4b8a-85c1-02dba5c65e4d.png)

So I'll resolve this now. :)

Reply:
Isn't it a problem that `DB_PORT` is used by Laravel? 🤔
So if I use `DB_PORT=54320` the container uses `54320` on the host machine and `5432` internally. Laravel uses `54320` but tries to connect via the docker internal network which would need `5432` as port.
Or am I wrong? 🤔

Shouldn't we use a dedicated `.env` (if possible) for docker? Or at least prefix all the Docker env with `DOCKER_`? So that it's `DB_PORT` for Laravel and `DOCKER_DB_PORT` for docker.

Reply:
All variables will be exported in container so internally also it will use that port .. i don't think we will have benefit in having extra env_docker file it will be same only

Reply:
@mallardduck could you please confirm this ?

Reply:
That's a great catch - I opted to stop my local postgres earlier.
But I just verified this is an issue and have a fix in the pipeline.

Sending another PR...

Reply:
So in the PR here: https://github.com/bhushan/opendor.me/pull/2/files

We just ensure the chosen port is used for exposing locally to the host AND internally on the container.

Reply:
Are you sure we need this @mallardduck ? I dont think we have any issue.. If yes then please do it for redis as well.. 

And also would like to know what issue we have 😄

Appreciate your efforts 🤘🏻

Reply:
All fixed in the PR now for both Postgres and Redis.
As mentioned in the PR @Gummibeer was right that this did cause errors.
Since between the containers postgress was still on the original port.

Reply:
What issue exactly?

Reply:
The problem is that if you change `DB_PORT` (which is used by Laravel and docker-compose) you change the port Laravel wants to use inside the container and the port the DB exposes to the host. But inside the DB port remains the default.
So what Dan changed is that the inside port is also changed to the same as outside. After that all exposed and expected ports are the same.

Reply:
I agree @mallardduck I was able to reproduce it, I have added comments, once comments are resolved I will approve and merge PR. THANX

Reply:
I've opted to close the second PR I sent so you can address it however you want.
So this concern is still open but is up for being solved in any manner you choose.

Reply:
Better 👍🏻